ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ContractKeysIT.scala

```diff
@@ -362,4 +362,38 @@ final class ContractKeysIT extends LedgerTestSuite {
         assertGrpcError(failedFetch, Status.Code.INVALID_ARGUMENT, "not visible")
       }
   })
+
+  test(
+    "CKDisclosedContractKeyReusability",
+    "Subsequent disclosed contracts can use the same contract key",
+    allocate(SingleParty, SingleParty),
+  )(implicit ec => {
+    case Participants(Participant(ledger1, party1), Participant(ledger2, party2)) =>
+      for {
+        // Create a helper contract and exercise a choice creating and disclosing a WithKey contract
+        creator1 <- ledger1.create(party1, WithKeyCreator(party1, party2))
+        withKey1 <- ledger1.exerciseAndGetContract[WithKey](
+          party1,
+          creator1.exerciseWithKeyCreator_DiscloseCreate(_, party1),
+        )
+
+        // Verify that the withKey1 contract is usable by the party2
+        fetcher <- ledger1.create(party1, WithKeyFetcher(party1, party2))
+        _ <- ledger2.exercise(party2, fetcher.exerciseWithKeyFetcher_Fetch(_, withKey1))
+
+        // Archive the disclosed contract
+        _ <- ledger1.exercise(party1, withKey1.exerciseWithKey_Archive(_))
+
```

Comment:
It would be interesting to fetch again here, to verify that the submission is rejected on-ledger in a failed completion. So even if `p2`'s participant thinks the contract is still active, the committer must reject the transaction.

Reply:
That was going to be a part of another test case but I can make this check here as well.

Reply:
Done.